src/components/Delegate/StakingOpportunities.tsx

```diff
@@ -73,10 +73,109 @@ export const ValidatorName = ({ moniker, isStaking }: ValidatorNameProps) => {
   )
 }
 
-export const StakingOpportunities = ({ assetId, opportunities }: StakingOpportunitiesProps) => {
-  const isLoaded = true
-  const isStaking = opportunities.some(x => x.cryptoAmount)
-  const assetSymbol = useAppSelector(state => selectAssetByCAIP19(state, assetId)).symbol
+export const StakingOpportunities = ({ assetId }: StakingOpportunitiesProps) => {
+  const validatorStatus = useAppSelector(selectStakingDataStatus)
+  const isLoaded = validatorStatus === 'loaded'
+  const asset = useAppSelector(state => selectAssetByCAIP19(state, assetId))
+  const marketData = useAppSelector(state => selectMarketDataById(state, assetId))
+  const dispatch = useAppDispatch()
+  const chainAdapterManager = useChainAdapters()
+  const [chainAdapter, setChainAdapter] = useState<ChainAdapter<ChainTypes> | null>(null)
+  const [address, setAddress] = useState<string>('')
+
+  const accountSpecifier = useMemo(() => {
+    if (!address.length) return ''
+
+    return caip10.toCAIP10({
+      caip2: asset.caip2,
+      account: address
+    })
+  }, [address, asset.caip2])
+
+  const {
+    state: { wallet }
+  } = useWallet()
```

Comment:
Note: I lifted this up in `<Staking />` router in https://github.com/shapeshift/web/pull/1357/ - this is totally fine for now but reading this, I'm wondering we might want to lift it even more up to not have this duplicated logic